src/libsyncengine/jobs/network/API_v2/upload_session/driveuploadsession.cpp

```diff
@@ -109,4 +111,26 @@ bool DriveUploadSession::handleCancelJobResult(const std::shared_ptr<UploadSessi
 
     return true;
 }
+
+void DriveUploadSession::abort() {
+    AbstractUploadSession::abort();
+    setVfsForceStatusCallback(nullptr);
+
+#ifdef __APPLE__
+    // Removing all extended attributes so that the file that failed to be uploaded is not identified as a dehydrated placeholder
+    // by the application when restarting the synchronization.
+
+    const SyncPath &localPath = getFilePath();
+    static const std::vector<const char *> attributes = {EXT_ATTR_STATUS, EXT_ATTR_PIN_STATE};
+
+    for (const auto attribute: attributes) {
+        if (auto ioError = IoError::Success;
+            !IoHelper::removeXAttr(localPath, EXT_ATTR_STATUS, ioError) || ioError != IoError::NoSuchFileOrDirectory) {
```

Comment:
**Questions.**
- Is a missing permission an actual problem for the user?
- In case of unexpected failure, should the exit code be changed?